### DISCUSSION TITLE
Selenium Vesrions for Sauce or WebdriverIO services-Need a clarification

### DISCUSSION THREAD

**Discussion by bajipogadadanda** [webdriverio/webdriverio#13081] [General]
Hi,
As I know Selenium standalone service can be used to define a particular selenium version(selenium 3 versus selenium4). 
If we use the Sauce service then will it use the latest selenium version( selenium4)? when I went through the sauce metadata logs , I found that the Selenium Version is null, but the webdriver protocol is W3C. So, let me know if Sauce service uses the selenium underhood, if yes which version, or since webdriver protocol is W3C, would it support all the selenium4 features?

**Comment by christian-bromann** [comment]
> As I know Selenium standalone service can be used to define a particular selenium version(selenium 3 versus selenium4).

That is not correct. WebdriverIO doesn't use Selenium. The service is mostly responsible for downloading and setting up browser driver for your tests.

> would it support all the selenium4 features?

What are Selenium 4 features you are looking for?

**Reply by bajipogadadanda** [reply]
Thanks Chris!
I am looking for selenium-4 features like Relative locators, Better Window/Tab Management(like new windows and tab handling, tab switching and e.t.c), and chaining of actions.
as I mentioned, I could see the web driver Protocol as  as W3C for Sauce service, is it the case if we use chromed-river service ? as we are using WebdriverIO Sauce and chrome services for execution, wanted to leverage Selenium4 features.

**Reply by christian-bromann** [reply]
> Relative locators, Better Window/Tab Management(like new windows and tab handling, tab switching and e.t.c), and chaining of actions.

These sound all like Selenium features which aren't supported by WebdriverIO.

**Reply by bajipogadadanda** [reply]
yes, these are selenium features, Chris. Since WebdriverIO supports W3C protocol for the concerned web drivers, I thought it supports most of the selenium 4 features. Thanks for the clarification.
I think  WDIO(with Mocha) supports some of these features differently, like using keywords swith Window and e.t.c for Window/Tab management.  Even relative locators also. I got below as some reference, I haven't tested it yet.


describe('Using relative locators in WebdriverIO', () => {
    it('should find elements relative to another element', async () => {
        await browser.url('https://example.com');

        // Find a reference element
        const referenceElement = await $('#reference');

        // Use relative locators to find an element below the reference element
        const relativeElement = await browser.$$('button=Cancel')[0].$(function () {
            return this.below(referenceElement);
        });

        expect(await relativeElement.isDisplayed()).to.be.true;
        await relativeElement.click();
    });
});

**Reply by christian-bromann** [reply]
> Even relative locators also.

WebdriverIO has no support for relative locators. `this.below` is not a thing in WebdriverIO.
